File: QuizAppServer/src/server/ServerMain.java
package server;

import database.FillUp;

import java.io.IOException;
import java.net.ServerSocket;
import java.util.ArrayList;

public class ServerMain {
    private static final String IP = "127.0.0.1";
    private static final int PORT = 5000;
    public static ArrayList<ClientThread> clientThreadPools;
    public static void main(String[] args) {
        //new FillUp().fillData();
        clientThreadPools = new ArrayList<>();
        try(ServerSocket serverSocket = new ServerSocket(PORT)) {
            while(true){
                ClientThread clientThread = new ClientThread(serverSocket.accept());
                //oppenentSearch(clientThread);
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public static void oppenentSearch(ClientThread clientThread) throws InterruptedException {

        for(int i=0; i<clientThreadPools.size(); i++){
            ClientThread opponent = clientThreadPools.get(i);
            if(opponent.isOpponentAssigned() ==false){
                RandQuestions randQuestions = new RandQuestions();
                opponent.setOpponentAssigned(true);
                opponent.setOpponetThread(clientThread);
                opponent.setRandQuestions(randQuestions);
                Thread.sleep(30);
                clientThread.setOpponentAssigned(true);
                clientThread.setOpponetThread(opponent);
                clientThread.setRandQuestions(randQuestions);

                break;
            }
        }

         clientThreadPools.add(clientThread);
    }
}

File: QuizAppServer/src/server/ServerMain.kt
package server

import java.io.IOException
import java.net.ServerSocket
import java.util.*

object ServerMain {
    private const val IP = "127.0.0.1"
    private const val PORT = 5000
    var clientThreadPools: ArrayList<ClientThread?>? = null

    @JvmStatic
    fun main(args: Array<String>) {
        //new FillUp().fillData();
        clientThreadPools = ArrayList()
        try {
            ServerSocket(PORT).use { serverSocket ->
                while (true) {
                    val clientThread = ClientThread(serverSocket.accept())
                    //oppenentSearch(clientThread);
                }
            }
        } catch (e: IOException) {
            e.printStackTrace()
        }
    }

    @Throws(InterruptedException::class)
    fun oppenentSearch(clientThread: ClientThread) {
        for (i in clientThreadPools!!.indices) {
            val opponent = clientThreadPools!![i]
            if (opponent!!.isOpponentAssigned == false) {
                val randQuestions = RandQuestions()
                opponent.isOpponentAssigned = true
                opponent.setOpponetThread(clientThread)
                opponent.setRandQuestions(randQuestions)
                Thread.sleep(30)
                clientThread.isOpponentAssigned = true
                clientThread.setOpponetThread(opponent)
                clientThread.setRandQuestions(randQuestions)
                break
            }
        }
        clientThreadPools!!.add(clientThread)
    }
}